packages/relic/lib/src/util/util.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:relic/src/map/shelf_unmodifiable_map.dart';
+
+/// Run [callback] and capture any errors that would otherwise be top-leveled.
+///
+/// If `this` is called in a non-root error zone, it will just run [callback]
+/// and return the result. Otherwise, it will capture any errors using
+/// [runZoned] and pass them to [onError].
+void catchTopLevelErrors(void Function() callback,
+    void Function(dynamic error, StackTrace) onError) {
+  if (Zone.current.inSameErrorZone(Zone.root)) {
+    return runZonedGuarded(callback, onError);
+  } else {
+    return callback();
+  }
+}
+
+/// Returns a [Map] with the values from [original] and the values from
+/// [updates].
+///
+/// For keys that are the same between [original] and [updates], the value in
+/// [updates] is used.
+///
+/// If [updates] is `null` or empty, [original] is returned unchanged.
+Map<K, V> updateMap<K, V>(Map<K, V> original, Map<K, V?>? updates) {
+  if (updates == null || updates.isEmpty) return original;
+
+  final value = Map.of(original);
+  for (var entry in updates.entries) {
+    final val = entry.value;
+    if (val == null) {
+      value.remove(entry.key);
+    } else {
+      value[entry.key] = val;
+    }
+  }
+
+  return value;
+}
+
+/// Adds a header with [name] and [value] to [headers], which may be null.
+///
+/// Returns a new map without modifying [headers].
+Map<String, Object> addHeader(
+  Map<String, Object>? headers,
```

Comment:
Is this used? If so it should be typed (`List<String>` instead of `Object`). If it's no longer used it should be removed.